src/setuptools_scm/hg.py

```diff
@@ -28,10 +29,13 @@ def _hg_tagdist_normalize_tagcommit(root, tag, dist, node, branch):
         return meta(tag)
 
 
-def parse(root):
+def parse(root, config=None):
```

Comment:
as i seeing this pattern i believe we should have a common pattern for this 

```python
def func(root, ...):
   ...
```
should replace with

```
def func(config_or_root=None, ..., root=None):
   config = Configuration.maybe_from_legacy_args(root_or_config, root)
```
(its possible i missed some calls where its not always possible that way)

this helper could trigger deprecationwarnings

Reply:
Hmm, the basic idea was to keep existing `func`s as they were, because someone could have implemented a different provider relying on the existing `parse` signature. Requiring adding something to parse would be a breaking change for them.

Since the introspection clicks on the named argument, root could be left out entirely since it's redundant. Not having positional args could be checked. Then again you considered git.parse as API to keep stable above.
```python
def func(config=, **kwargs)
```

Reply:
i see, i'm deferring handling this for the future then